### PR TITLE
feat(bigquery_execute): add destination_table for query-to-table jobs

### PR DESCRIPTION
**Pull request by jatorre**
## What

Adds optional `destination_table`, `write_disposition`, and `create_disposition` named parameters to `bigquery_execute`. When `destination_table` is set, the query result is materialised into that table.

```sql
SELECT * FROM bigquery_execute(
  'my-project',
  'SELECT id, ST_GEOGFROMTEXT(wkt) AS geom FROM staging',
  destination_table := 'dataset.table',
  write_disposition := 'WRITE_TRUNCATE'   -- default; also WRITE_APPEND / WRITE_EMPTY
);
```

## Why

`bigquery_execute` runs everything through the `jobs.query` REST endpoint, which **cannot write to a destination table** — there's no way to express "run this SELECT and land the result in `dataset.table`".

A concrete need: converting a staged WKT column into a native `GEOGRAPHY` with `SELECT ... ST_GEOGFROMTEXT(...)` that writes into the destination using `WRITE_TRUNCATE`. `WRITE_TRUNCATE` preserves the table's partitioning/clustering spec on re-runs, which `CREATE OR REPLACE TABLE AS SELECT` drops — and which `jobs.query` can't do regardless.

## How

When `destination_table` is present, the query is submitted as a **`jobs.insert`** job carrying a `JobConfigurationQuery` (destination table + write/create disposition), then polled to completion via `WaitForJobCompletion`. This mirrors the existing `bigquery_load` `JobConfigurationLoad` path almost exactly (`BuildQueryJobRequest` is the analogue of `BuildLoadJobRequest`), and reuses the `timeout_ms` plumbing from #195 (nullopt waits to completion).

- New `BigqueryClient::ExecuteQueryToTable(...)` + private `BuildQueryJobRequest(...)`.
- `bigquery_execute` parses the three new params; absent `destination_table`, behaviour is unchanged (still `jobs.query`).
- `destination_table` accepts `dataset.table` or `project.dataset.table` (backticks tolerated); project falls back to the connection's project.
- `destination_table` + `dry_run` is rejected.

Rebased onto `main` after #195 — that PR's query-job polling and per-call `timeout_ms` are complementary (they fixed inline-wait timeouts for `jobs.query`); this PR adds the orthogonal *destination-table* capability and plugs into the same `timeout_ms` parameter.

## Verified

Built locally (DuckDB v1.5.3, arm64-osx, ADC) and run end-to-end against BigQuery:
- `destination_table` write + `ST_GEOGFROMTEXT` GEOGRAPHY round-trip ✅
- idempotent `WRITE_TRUNCATE` re-run (table + its spec preserved) ✅
- heavy 2B-row aggregation materialised into a table, polled to completion (`2000000000` landed) ✅

### CODE REVIEW

**Comment by jatorre** [comment]
Verified end-to-end against BigQuery with a local build (DuckDB v1.5.3, arm64-osx, ADC creds).

**1. Functional — `destination_table` write + GEOGRAPHY round-trip + idempotent `WRITE_TRUNCATE`:**

```
SELECT * FROM bigquery_execute('proj',
  'SELECT 1 AS id, ST_GEOGFROMTEXT(''POINT(-3.7 40.4)'') AS geom',
  destination_table := 'ds.tbl', write_disposition := 'WRITE_TRUNCATE');
-- success=true, job_duckdb_query_*
SELECT id, ST_ASTEXT(geom) FROM ds.tbl;  -- 1 | POINT(-3.7 40.4)
```
Re-running with a different row replaces contents in place (table — and its partitioning/clustering — preserved), confirming `WRITE_TRUNCATE` semantics rather than drop-and-recreate.

**2. The motivating case — survives a timeout that kills the inline path.** With `SET bq_query_timeout_ms = 1` and a deliberately heavy 2B-row aggregation:

| path | result |
|------|--------|
| `bigquery_execute(...)` (jobs.query, inline wait) | `Query execution exceeded the timeout. Job ID: ...` ❌ |
| `bigquery_execute(..., destination_table := ...)` (jobs.insert + `WaitForJobCompletion`) | success ✅ — `2000000000` landed in the table |

Same query, same 1 ms ceiling — the destination path polls the job to completion instead of failing inline. Marking ready for review.

**Comment by jatorre** [comment]
Rebased onto `main` after #195 merged. #195 already fixed the inline-wait timeout for `jobs.query` and added a per-call `timeout_ms`, so this PR is now scoped purely to the orthogonal capability: **materialising a query result into a destination table** (which `jobs.query` can't express at all). It reuses #195's `timeout_ms` plumbing — `destination_table` jobs poll via `WaitForJobCompletion(timeout_ms)`, `nullopt` waiting to completion.

Re-verified on the rebased code against BigQuery: GEOGRAPHY round-trip, idempotent `WRITE_TRUNCATE`, and a heavy 2B-row aggregation polled to completion into a table. (Please disregard the 1 ms-timeout contrast in my earlier comment — that distinction is moot now that #195 unified timeout handling.)

File: src/bigquery_client.cpp [hunk 1]
```diff
@@ -798,6 +798,53 @@ google::cloud::bigquery::v2::Job BigqueryClient::LoadParquetUris(const BigqueryT
     return WaitForJobCompletion(response->job_reference(), timeout_ms);
 }
 
+google::cloud::bigquery::v2::Job BigqueryClient::ExecuteQueryToTable(const string &query,
+                                                                    const BigqueryTableRef &destination_table,
+                                                                    const string &write_disposition,
+                                                                    const string &create_disposition,
+                                                                    const string &location,
+                                                                    const std::map<string, string> &labels,
+                                                                    const std::optional<int> &timeout_ms) {
```
Comment (hafenkran) [inline src/bigquery_client.cpp:807]: Right now the `bq_query_timeout_ms` session timeout is ingored here. It passes `data.timeout_ms` directly to `ExecuteQueryToTable`. If the caller only set `bq_query_timeout_ms`, `data.timeout_ms` is `nullopt`, and `WaitForJobCompletion` treats that as wait forever. This breaks the documented behavior

File: src/bigquery_execute.cpp [hunk 2]
```diff
@@ -96,6 +171,31 @@ static void BigQueryExecuteFunc(ClientContext &context, TableFunctionInput &data
...
+        data.finished = true;
+
+        const auto &job_ref = job.job_reference();
+        output.SetValue(0, 0, true);
+        output.SetValue(1, 0, Value(job_ref.job_id()));
+        output.SetValue(2, 0, Value(job_ref.project_id()));
+        output.SetValue(3, 0, job_ref.has_location() ? Value(job_ref.location().value()) : Value(LogicalType::VARCHAR));
+        output.SetValue(4, 0, Value::UBIGINT(0));
+        output.SetValue(5, 0, Value::BIGINT(0));
+        output.SetValue(6, 0, Value(LogicalType::VARCHAR));
```
Comment (hafenkran) [inline src/bigquery_execute.cpp]: This sets total_rows = 0, total_bytes_processed = 0, and num_dml_affected_rows = NULL for every destination-table query. BigQuery job results expose query stats via job.statistics().query(); the repo already reads total_bytes_processed from there for the other run.

File: src/bigquery_client.cpp [hunk 3]
```diff
@@ -798,6 +798,53 @@ google::cloud::bigquery::v2::Job BigqueryClient::LoadParquetUris(const BigqueryT
...
+    }
+
+    if (!response->has_job_reference()) {
+        throw BinderException("Query job submission succeeded but did not return a job reference");
+    }
+    if (response->has_status() && response->status().state() == "DONE") {
+        ThrowOnJobStatusError(response->status(), "Query job");
+        return *response;
+    }
+    return WaitForJobCompletion(response->job_reference(), timeout_ms);
```
Comment (hafenkran) [inline src/bigquery_client.cpp]: This reuses WaitForJobCompletion, but that helper hard-codes "Load job" in timeout/status errors. Failed destination queries can therefore surface as Load job failed, which points users at the wrong path.

File: src/bigquery_execute.cpp [hunk 4]
```diff
@@ -125,6 +225,9 @@ BigQueryExecuteFunction::BigQueryExecuteFunction()
     named_parameters["grpc_endpoint"] = LogicalType::VARCHAR;
     named_parameters["dry_run"] = LogicalType::BOOLEAN;
     named_parameters["timeout_ms"] = LogicalType::BIGINT;
+    named_parameters["destination_table"] = LogicalType::VARCHAR;
+    named_parameters["write_disposition"] = LogicalType::VARCHAR;
+    named_parameters["create_disposition"] = LogicalType::VARCHAR;
```
Comment (hafenkran) [inline src/bigquery_execute.cpp:240]: The PR adds destination_table, write_disposition, and create_disposition, but test/sql/functions/function_bigquery_execute.test is unchanged. Add a SQLLogicTest covering a successful destination_table write, invalid disposition, and destination_table + dry_run rejection.

Comment (hafenkran) [reply src/bigquery_execute.cpp:240]: The README’s bigquery_execute named-parameter table still ends at timeout_ms, so the new user-facing parameters are undocumented.

**Review by hafenkran — CHANGES_REQUESTED** [review]
Thanks for the PR! Here are some change requests

**Comment by hafenkran** [comment]
Thanks @jatorre for the PR, I like this direction and the feature makes sense. Requesting changes for now because a few edge cases need tightening up around timeout behavior, returned metadata, error reporting, tests, and docs.

File: src/bigquery_client.cpp (hunk 1 above)
Comment (jatorre) [reply src/bigquery_client.cpp:807]: Fixed in f3b0941. `ExecuteQueryToTable` now resolves the timeout the same way `ExecuteQuery` does — when no per-call `timeout_ms` is given it falls back to `BigquerySettings::QueryTimeoutMs()` (i.e. `bq_query_timeout_ms`) instead of passing `nullopt` through to `WaitForJobCompletion`. Verified live with `SET bq_query_timeout_ms = 1` against a heavy uncacheable aggregation: it now fails with `Query job execution exceeded the timeout` instead of waiting indefinitely.

File: src/bigquery_execute.cpp (hunk 2 above)
Comment (jatorre) [reply src/bigquery_execute.cpp]: Fixed in f3b0941. The destination path now reports real metadata: `total_bytes_processed` and `num_dml_affected_rows` come from `job.statistics().query()`, and `total_rows` is read via `GetQueryResults(job_ref)` (same source as the inline jobs.query path, since query-job statistics don't carry a result row count — that's load-job only). Anything BigQuery doesn't report stays NULL rather than a fake 0. Verified live: a `GROUP BY ... LIMIT 50` into a table returned `total_rows=50`, `total_bytes_processed=9170960028`, `num_dml_affected_rows=NULL`.

File: src/bigquery_client.cpp (hunk 3 above)
Comment (jatorre) [reply src/bigquery_client.cpp]: Fixed in f3b0941. `WaitForJobCompletion`/`GetLoadJobForCompletion` now take a `job_kind` argument (default `"Load job"`, so the load paths are unchanged), and `ExecuteQueryToTable` passes `"Query job"`. Failed/timed-out destination queries now surface as `Query job ...`. Verified live (see the timeout repro above).

File: src/bigquery_execute.cpp (hunk 4 above)
Comment (jatorre) [reply src/bigquery_execute.cpp:240]: Added in f3b0941 to `test/sql/functions/function_bigquery_execute.test`: a successful `destination_table` write (asserting `total_rows` and that the rows land via the attached catalog), an idempotent `WRITE_TRUNCATE` re-run, an invalid `write_disposition` rejection, and a `destination_table` + `dry_run` rejection. All 29 assertions pass against live BigQuery.

Comment (jatorre) [reply src/bigquery_execute.cpp:240]: Documented in f3b0941. The `bigquery_execute` parameter table now lists `destination_table`, `write_disposition`, and `create_disposition`, plus a short example of writing a query result into a table.

**Comment by jatorre** [comment]
Thanks for the thorough review @hafenkran — all five points addressed in f3b0941:

1. **Timeout** — `ExecuteQueryToTable` now falls back to `bq_query_timeout_ms` when no per-call `timeout_ms` is set, instead of waiting forever.
2. **Metadata** — `total_rows` (via `GetQueryResults`), `total_bytes_processed` and `num_dml_affected_rows` (via `job.statistics().query()`) are now real; un-reported values are NULL rather than 0.
3. **Error label** — query-job failures/timeouts now say `Query job …` (threaded a `job_kind` through `WaitForJobCompletion`; load paths keep the `"Load job"` default).
4. **Tests** — added SQLLogicTests for a successful write, `WRITE_TRUNCATE` idempotency, invalid disposition, and `destination_table` + `dry_run` rejection.
5. **Docs** — README parameter table + example updated.

Re-verified end-to-end against BigQuery (`make release`, full test file = 29 assertions passing; plus manual timeout/label and metadata checks).

**Comment by jatorre** [comment]
Follow-up cleanup in 292ff3c: while addressing the metadata point I noticed the `num_dml_affected_rows` result column was declared `VARCHAR` but always fed a `BIGINT` value (cast implicitly on output), inconsistent with the `total_rows`/`total_bytes_processed` integer columns beside it. Declared it `BIGINT` to match the value and updated the README example. This touches both the inline and destination paths consistently. Rebuilt and re-verified — 29/29 assertions pass against live BigQuery.

**Comment by hafenkran** [comment]
Thank you, LGTM!
